### PR TITLE
More Toxification

### PR DESCRIPTION
**Pull request by detiber**
- Move pylint tests to tox
- Move yamllint tests to tox
- Create separate tox config (and setup.py) for root
- bump ansible requirement
- unify pylint config
- add docs
- remove git directory containing old testing tools

### CODE REVIEW

**Comment by detiber** [comment]
@tbielawa need moar tox

**Comment by detiber** [comment]
aos-ci-test

**Comment by detiber** [comment]
aos-ci-test

**Comment by detiber** [comment]
aos-ci-test

**Comment by tbielawa** [comment]
@detiber wrt this PR and the [Use tox and molecule...](https://github.com/openshift/openshift-ansible/pull/2998) issue. The other issue was submited first, does this PR depend on that PR?

**Comment by detiber** [comment]
@tbielawa no, this one breaks out a portion of the tox and molecule pr without introducing molecule.

**Comment by detiber** [comment]
aos-ci-test

**Comment by detiber** [comment]
aos-ci-test

**Comment by detiber** [comment]
aos-ci-test

**Comment by detiber** [comment]
aos-ci-test

**Comment by detiber** [comment]
@tbielawa ptal, this should be ready to go now.

**Comment by detiber** [comment]
@sdodson ptal

File: utils/test/openshift_ansible_tests.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    def test_write_inventory_vars_role_vars(self):
+        print(yaml.dump(openshift_ansible.CFG.deployment.roles))
+        with open(self.inventory, 'w') as inv:
+            openshift_ansible.CFG.deployment.roles['master'].variables={'color': 'blue'}
+            openshift_ansible.CFG.deployment.roles['node'].variables={'color': 'green'}
+            openshift_ansible.write_inventory_vars(inv, None)
+
+        inventory = configparser.ConfigParser(allow_no_value=True)
+        inventory.read(self.inventory)
+        print(inventory.sections())
```
Comment (ashcrow) [inline utils/test/openshift_ansible_tests.py]: Left over from debugging?

**Comment by ashcrow** [comment]
:hugs: for ``tox``!!

File: utils/test/openshift_ansible_tests.py (hunk 1 above)
Comment (sdodson) [inline utils/test/openshift_ansible_tests.py]: We can clean this up later if so, going ahead with merge.

**Comment by detiber** [comment]
@ashcrow @sdodson fix for debug statement is here: https://github.com/openshift/openshift-ansible/pull/3066

**Pull request merged**